src/crimson/net/ProtocolV2.cc

```diff
@@ -1433,6 +1445,14 @@ seastar::future<> ProtocolV2::read_message(utime_t throttle_stamp)
                            0};
     ceph_msg_footer footer{0, 0, 0, 0, current_header.flags};
 
+    // TODO: large chunks can be fragmented even while using the POSIX stack
+    // and being placed on the same, huge memory block.
+    // We need to merge them.
+    logger().debug("msg_frame.data()={}", msg_frame.data());
+    using ceph::msgr::v2::segment_t;
+    ceph_assert(msg_frame.data().get_num_buffers() <= 1);
```

Comment:
**These asserts are just for testing**. We'll leave the right to decide about SGL's structure to network stack. For the DPDK-based one the `bufferlist` will likely be very fragmented and that's fine.